File: Car.scala
class Car(val year: Int, var mileage: Int){
  def drive(distance: Int) = {
    mileage += distance
  }

  def turn(direction: String) = {
    println("turning" + " " + direction)
  }
}

val car = new Car(1990, 0)
car.turn("left")
car turn "left"

File: Car.java
public class Car {
    final private int year;
    private int mileage;

    public Car(int year){
        this.year = year;
        this.mileage = 0;
    }

    public int getYear(){
        return year;
    }

    public int getMileage(){
        return mileage;
    }

    public void drive(int distance){
        this.mileage += distance;
    }

    public void turn(String direction){
        System.out.println("turning" + " " + direction);
    }

    public String toString() {
        return this.year + " " + this.mileage;
    }
}
